components/list/list-item-expand-collapse-mixin.js

```diff
@@ -0,0 +1,172 @@
+
+	constructor() {
+		super();
+		this._siblingHasNestedItems = false;
+		this._renderExpandCollapseSlot = false;
+		this._showNestedLoadingSpinner = false;
+		this._parentChildUpdateSubscription = new EventSubscriberController(this, {}, { eventName: 'd2l-list-child-status' });
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		// mixin requires key for events
+		if (!this.key && this.expandable) {
+			console.warn('ListItemExpandCollapseMixin requires a key.');
+			this.expandable = false;
+		}
+	}
+
+	updated(changedProperties) {
+		if (changedProperties.has('_siblingHasNestedItems') || changedProperties.has('expandable')) {
+			this._renderExpandCollapseSlot = this.expandable || this._siblingHasNestedItems;
+		}
+		if (changedProperties.has('_draggingOver') && this._draggingOver && this.dropNested && !this.expanded && this.expandable) {
+			let elapsedHoverTime = 0;
+			let dragIntervalId = null;
+			const watchDraggingOver = () => {
+				if (elapsedHoverTime >= dragHoverDropTime) {
+					if (this._draggingOver) {
+						this._toggleExpandCollapse();
+					}
+					clearInterval(dragIntervalId);
+				} else {
+					if (!this._draggingOver) {
+						clearInterval(dragIntervalId);
+					} else {
+						elapsedHoverTime += dragIntervalDelay;
+					}
+				}
+			};
+			// check if they are still hovered over same item every 100ms
+			dragIntervalId = setInterval(watchDraggingOver, dragIntervalDelay);
+		}
+		if (changedProperties.has('expanded') || changedProperties.has('_hasNestedList')) {
+			this._showNestedLoadingSpinner = this.expanded && !this._hasNestedList;
+		}
+	}
+
+	updateSiblingHasChildren(siblingHasNestedItems) {
+		this._siblingHasNestedItems = siblingHasNestedItems;
+	}
+
+	_renderExpandCollapse() {
+		if (!this.expandable) {
+			return nothing;
+		}
+		return html`
+			<d2l-button-icon
+				icon="${this.expanded ? 'tier1:arrow-collapse-small' : 'tier1:arrow-expand-small' }"
+				aria-expanded="${this.expanded ? 'true' : 'false'}"
+				text="${this.label}"
+				@click="${this._toggleExpandCollapse}"></d2l-button-icon>`;
+	}
+
+	_renderExpandCollapseAction() {
+		if (this.selectable || !this.expandable || this.noPrimaryAction) {
+			return nothing;
+		}
+
+		return html`<div class="d2l-list-expand-collapse-action" @click="${this._toggleExpandCollapse}"></div>`;
+	}
+
+	_renderNestedLoadingSpinner() {
+		if (!this.expandable || !this._showNestedLoadingSpinner) {
+			return nothing;
+		}
+		return html`
+			<div class="d2l-list-nested-loading">
+				<d2l-loading-spinner size="40"></d2l-loading-spinner>
+			</div>`;
+	}
+
+	_toggleExpandCollapse(e = null) {
+		if (e) {
+			e.stopPropagation();
+		}
+		if (!this.expandable) {
+			return;
+		}
+		this.expanded = !this.expanded;
+		/** Dispatched whenever the list item expand state is toggled. Dispatched with the list item key, previous expanded state, and current expanded state. */
+		this.dispatchEvent(new CustomEvent('d2l-list-item-expand-collapse-toggled', {
+			detail: {
```

Comment:
Is this event detail needed? Looks like the consumer can access all this by:
```javascript
key = e.target.key;
oldState = !e.target.expanded;
newState = e.target.expanded;
```

Reply:
Good point, it might not be necessary. I'll test it out!

Reply:
Works as expected - I'll push these changes.